File: src/Http.API/ClientApp/src/app/share/admin/services/third-news.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { ThirdNewsFilterDto } from '../models/third-news/third-news-filter-dto.model';
import { ThirdNewsAddDto } from '../models/third-news/third-news-add-dto.model';
import { ThirdNewsUpdateDto } from '../models/third-news/third-news-update-dto.model';
import { ThirdNewsBatchUpdateDto } from '../models/third-news/third-news-batch-update-dto.model';
import { ThirdNewsItemDtoPageList } from '../models/third-news/third-news-item-dto-page-list.model';
import { ThirdNews } from '../models/third-news/third-news.model';

/**
 * 资讯管理
 */
@Injectable({ providedIn: 'root' })
export class ThirdNewsService extends BaseService {
  /**
   * 筛选
   * @param data ThirdNewsFilterDto
   */
  filter(data: ThirdNewsFilterDto): Observable<ThirdNewsItemDtoPageList> {
    const url = `/api/admin/ThirdNews/filter`;
    return this.request<ThirdNewsItemDtoPageList>('post', url, data);
  }

  /**
   * 新增
   * @param data ThirdNewsAddDto
   */
  add(data: ThirdNewsAddDto): Observable<ThirdNews> {
    const url = `/api/admin/ThirdNews`;
    return this.request<ThirdNews>('post', url, data);
  }

  /**
   * 更新
   * @param id 
   * @param data ThirdNewsUpdateDto
   */
  update(id: string, data: ThirdNewsUpdateDto): Observable<ThirdNews> {
    const url = `/api/admin/ThirdNews/${id}`;
    return this.request<ThirdNews>('put', url, data);
  }

  /**
   * 详情
   * @param id 
   */
  getDetail(id: string): Observable<ThirdNews> {
    const url = `/api/admin/ThirdNews/${id}`;
    return this.request<ThirdNews>('get', url);
  }

  /**
   * ⚠删除
   * @param id 
   */
  delete(id: string): Observable<ThirdNews> {
    const url = `/api/admin/ThirdNews/${id}`;
    return this.request<ThirdNews>('delete', url);
  }

  /**
   * 批量操作
   * @param data ThirdNewsBatchUpdateDto
   */
  batchUpdate(data: ThirdNewsBatchUpdateDto): Observable<boolean> {
    const url = `/api/admin/ThirdNews/batchUpdate`;
    return this.request<boolean>('put', url, data);
  }

}

File: src/Sdks/Dusi.Manage.Client/Services/ThirdNewsService.cs
using Core.Entities.CMS;
using Share.Models.ThirdNewsDtos;
namespace Dusi.Manage.Client.Services;
/// <summary>
/// 资讯管理
/// </summary>
public class ThirdNewsService : BaseService
{
    public ThirdNewsService(HttpClient httpClient) : base(httpClient)
    {

    }
    /// <summary>
    /// 筛选
    /// </summary>
    /// <param name="data">ThirdNewsFilterDto</param>
    /// <returns></returns>
    public async Task<PageList<ThirdNewsItemDto>?> FilterAsync(ThirdNewsFilterDto data)
    {
        string url = $"/api/admin/ThirdNews/filter";
        return await PostJsonAsync<PageList<ThirdNewsItemDto>?>(url, data);
    }

    /// <summary>
    /// 新增
    /// </summary>
    /// <param name="data">ThirdNewsAddDto</param>
    /// <returns></returns>
    public async Task<ThirdNews?> AddAsync(ThirdNewsAddDto data)
    {
        string url = $"/api/admin/ThirdNews";
        return await PostJsonAsync<ThirdNews?>(url, data);
    }

    /// <summary>
    /// 更新
    /// </summary>
    /// <param name="id"> </param>
    /// <param name="data">ThirdNewsUpdateDto</param>
    /// <returns></returns>
    public async Task<ThirdNews?> UpdateAsync(string id, ThirdNewsUpdateDto data)
    {
        string url = $"/api/admin/ThirdNews/{id}";
        return await PutJsonAsync<ThirdNews?>(url, data);
    }

    /// <summary>
    /// 详情
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<ThirdNews?> GetDetailAsync(string id)
    {
        string url = $"/api/admin/ThirdNews/{id}";
        return await GetJsonAsync<ThirdNews?>(url);
    }

    /// <summary>
    /// ⚠删除
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<ThirdNews?> DeleteAsync(string id)
    {
        string url = $"/api/admin/ThirdNews/{id}";
        return await DeleteJsonAsync<ThirdNews?>(url);
    }

    /// <summary>
    /// 批量操作
    /// </summary>
    /// <param name="data">ThirdNewsBatchUpdateDto</param>
    /// <returns></returns>
    public async Task<bool?> BatchUpdateAsync(ThirdNewsBatchUpdateDto data)
    {
        string url = $"/api/admin/ThirdNews/batchUpdate";
        return await PutJsonAsync<bool?>(url, data);
    }

}